lib/stackprof/middleware.rb

```diff
@@ -11,7 +11,8 @@ def initialize(app, options = {})
       Middleware.interval = options[:interval] || 1000
       Middleware.raw      = options[:raw] || false
       Middleware.enabled  = options[:enabled]
-      Middleware.path     = options[:path] || 'tmp'
+      Middleware.out      = options[:out] || nil
+      Middleware.path     = 'tmp'
```

Comment:
Looks like the path parameter got lost?

Reply:
I intentionally did this because the path would be either extracted from the `out` option or default to `tmp`. If the user specifies a file only (no path, like in `out: 'my_file.dump'`), `File.dirname("my_file.dump")` will return `"."` which will cause the file to be stored the in the project's root directory (`./my_file.dump`).

Reply:
In other words, `tmp` will only be used if specified directly in the path passed to `out` or when the `out` option is not passed at all.

Reply:
Hmm I see. I'm worried about breaking backwards compatibility. If someone was using only path before, it would break after upgrading.

Reply:
Ah, that's a good point. The other reason that made me remove `path` was that I didn't want to keep two overlapping options, but now that you mention this, a decision needs to be made on whether to just modify the way `path` was being set, or to add `out` option alongside. I also noticed that there are no references to the middleware's `path` option in the README, so unless someone read the code of that file and decided to use the `path` option, they wouldn't know it exists.

Reply:
The `out` option is also consistent with the one available for Ruby in the `StackProf.run` method.

Reply:
How about we check if path is a directory, and if not we treat it as a full file name? Then we don't need two options.

Reply:
That works for me. Should I go ahead and make that change?

Reply:
Yes, please